core-bundle/src/DependencyInjection/ContaoCoreExtension.php

```diff
@@ -95,6 +97,28 @@ public function load(array $configs, ContainerBuilder $container): void
         ;
     }
 
+    private function handleSearchIndexer(array $config, ContainerBuilder $container): void
+    {
+        $container
+            ->registerForAutoconfiguration(IndexerInterface::class)
+            ->addTag('contao.search_indexer')
+        ;
+
+        // Set the two parameters so they can be used in our legacy Config class for maximum BC
+        $container->setParameter('contao.search.default_indexer.enable', $config['search']['default_indexer']['enable']);
+        $container->setParameter('contao.search.default_indexer.enableIndexProtected', $config['search']['default_indexer']['enableIndexProtected']);
+
+        $defaultIndexer = $container->getDefinition('contao.search.indexer.default');
+
+        // Configure whether to index protected pages on the default indexer
+        $defaultIndexer->setArgument(1, $config['search']['default_indexer']['enableIndexProtected']);
+
+        // Remove the default indexer completely if it was disabled
```

Comment:
We should probably check this at the beginning and return early?

Reply:
I didn‘t do that on purpose. I like it better that way🤷‍♂️ The only thing that could be moved is the argument thing anyway.